autoarray/inversion/inversion/interferometer.py

```diff
@@ -208,11 +212,11 @@ def __init__(
     @classmethod
     def from_data_mapper_and_regularization(
         cls,
-        visibilities: vis.Visibilities,
-        noise_map: vis.VisibilitiesNoiseMap,
-        transformer: trans.TransformerNUFFT,
-        mapper: typing.Union[mappers.MapperRectangular, mappers.MapperVoronoi],
-        regularization: reg.Regularization,
+        visibilities: Visibilities,
+        noise_map: VisibilitiesNoiseMap,
+        transformer: TransformerNUFFT,
+        mapper: Union[MapperRectangular, MapperVoronoi],

```

Comment:
If these both inherit from some common Mapper class then it would be best to use that type instead of Union.

If there are methods both implement which are used here then those should be defined as abstractmethods in the common parent class.